src/config/mod.rs

```diff
@@ -223,6 +227,11 @@ pub struct SourceOuter {
     pub acknowledgements: bool,
     #[serde(flatten)]
     pub(super) inner: Box<dyn SourceConfig>,
+    #[serde(
+        default,
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    pub proxy: ProxyConfig,
```

Comment:
It probably doesn't matter to serde, but I would prefer to leave `inner` as the last element in this struct.